# Analisis del Desempeño de Empleados

## Librerias

In [3]:
# librerias
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

## Carga de datos

In [4]:
# Estilo de graficas
sns.set_theme(style="whitegrid")

In [5]:
# Carga de datos
data = pd.read_csv('./employees_final_dataset.csv')
data

employee_id         department     region  education gender  \
0             8724         Technology  region_26  Bachelors      m   
1            74430                 HR   region_4  Bachelors      f   
2            72255  Sales & Marketing  region_13  Bachelors      m   
3            38562        Procurement   region_2  Bachelors      f   
4            64486            Finance  region_29  Bachelors      m   
...            ...                ...        ...        ...    ...   
17412        64573         Technology   region_7  Bachelors      f   
17413        49584                 HR   region_7  Bachelors      m   
17414        49584                 HR   region_7  Bachelors      m   
17415        49584                 HR   region_7  Bachelors      m   
17416        64573                 HR   region_7  Bachelors      f   

      recruitment_channel  no_of_trainings  age  previous_year_rating  \
0                sourcing                1   24                   NaN   
1                   other                1   31                   3.0   
2                   other                1   31                   1.0   
3                   other                3   31                   2.0   
4                sourcing                1   30                   4.0   
...                   ...              ...  ...                   ...   
17412            referred                2   30                   5.0   
17413               other                1   33                   1.0   
17414               other                1   33                   1.0   
17415               other                1   33                   1.0   
17416            referred                1   35                   5.0   

       length_of_service  KPIs_met_more_than_80  awards_won  \
0                      1                      1           0   
1                      5                      0           0   
2                      4                      0           0   
3                      9                      0           0   
4                      7                      0           0   
...                  ...                    ...         ...   
17412                  6                      1           0   
17413                  9                      0           0   
17414                  9                      0           0   
17415                  9                      0           0   
17416                  4                      1           0   

       avg_training_score  
0                      77  
1                      51  
2                      47  
3                      65  
4                      61  
...                   ...  
17412                  81  
17413                  51  
17414                  51  
17415                  51  
17416                  81  

[17417 rows x 13 columns]

In [6]:
# eliminamos la columna employee_id
data = data.drop('employee_id', axis=1)

# Creamos una copia
dataset = data.copy()
dataset

department     region  education gender recruitment_channel  \
0             Technology  region_26  Bachelors      m            sourcing   
1                     HR   region_4  Bachelors      f               other   
2      Sales & Marketing  region_13  Bachelors      m               other   
3            Procurement   region_2  Bachelors      f               other   
4                Finance  region_29  Bachelors      m            sourcing   
...                  ...        ...        ...    ...                 ...   
17412         Technology   region_7  Bachelors      f            referred   
17413                 HR   region_7  Bachelors      m               other   
17414                 HR   region_7  Bachelors      m               other   
17415                 HR   region_7  Bachelors      m               other   
17416                 HR   region_7  Bachelors      f            referred   

       no_of_trainings  age  previous_year_rating  length_of_service  \
0                    1   24                   NaN                  1   
1                    1   31                   3.0                  5   
2                    1   31                   1.0                  4   
3                    3   31                   2.0                  9   
4                    1   30                   4.0                  7   
...                ...  ...                   ...                ...   
17412                2   30                   5.0                  6   
17413                1   33                   1.0                  9   
17414                1   33                   1.0                  9   
17415                1   33                   1.0                  9   
17416                1   35                   5.0                  4   

       KPIs_met_more_than_80  awards_won  avg_training_score  
0                          1           0                  77  
1                          0           0                  51  
2                          0           0                  47  
3                          0           0                  65  
4                          0           0                  61  
...                      ...         ...                 ...  
17412                      1           0                  81  
17413                      0           0                  51  
17414                      0           0                  51  
17415                      0           0                  51  
17416                      1           0                  81  

[17417 rows x 12 columns]

## Analisis de valores faltantes

In [7]:
# Codigo para correr las funciones de utils.ipynb
%run utils.ipynb

In [8]:
# Registros duplicados
print("Registros Duplicados: " ,data.duplicated().sum())

# Separador
print("="*40)

# Tamano de la data
print("Las dimensiones de la data son: ", data.shape)

Registros Duplicados:  18
Las dimensiones de la data son:  (17417, 12)


In [10]:
# Cantidad de valores completos
v_complet = dataset.missing.number_complete()
v_complet

206870

In [11]:
# Cantidad de valores faltantes
v_missing = dataset.missing.number_missing()
v_missing

2134

In [12]:
# Porcentaje de valores faltantes en el dataframe
pct_v = (v_missing/v_complet)*100
pct_v

1.0315657176004254

El porcentaje de valores faltantes es de 1.03% lo cual es un valor bastante bajo y por tanto podemos aplicar un proceso de eliminacion en los datos